### PR TITLE
DBZ-6227 Reduce CPU usage and ensure critical logs have task Uid

### PR DESCRIPTION
**Pull request by nancyxu123**
Remove Deadlock in BufferedPublisher, increase sleep time in some threads, and ensure critical logs have taskUid.

### CODE REVIEW

File: src/main/java/io/debezium/connector/spanner/kafka/internal/TaskSyncPublisher.java [hunk 1]
```diff
@@ -91,7 +91,6 @@ private void publishSyncEvent(TaskSyncEvent taskSyncEvent) {
             lastTime = Instant.now();
         }
         catch (ExecutionException e) {
-            close();
```
Comment (shantanu-sharechat) [inline src/main/java/io/debezium/connector/spanner/kafka/internal/TaskSyncPublisher.java:94]: How will Buffered publisher and this thread terminate if close is not called?

File: src/main/java/io/debezium/connector/spanner/db/stream/SpannerChangeStreamService.java [hunk 2]
```diff
@@ -83,6 +85,9 @@ public void getEvents(Partition partition, ChangeStreamEventConsumer changeStrea
                 start = now();
             }
         }
+        finally {
+            LOGGER.info("Task {}, Stopped streaming from partition", taskUid, token);
```
Comment (shantanu-sharechat) [inline src/main/java/io/debezium/connector/spanner/db/stream/SpannerChangeStreamService.java:89]: A place holder is missing.

File: src/main/java/io/debezium/connector/spanner/kafka/internal/TaskSyncPublisher.java (hunk 1 above)
Comment (nancyxu123) [reply src/main/java/io/debezium/connector/spanner/kafka/internal/TaskSyncPublisher.java:94]: Hi Shantanu, if there is an ExecutionException here, the error handler will have accepted this exception.

The error handler sets the exception to true the ChangeEventQueue by calling setProducerThrowable: https://github.com/debezium/debezium-connector-spanner/blob/e61c7446f4e79ba6a2432064968964f12419eb71/src/main/java/io/debezium/connector/spanner/task/SynchronizationTaskContext.java#L230 

If the queue's producer exception is present, we will throw an exception when calling poll(): https://github.com/debezium/debezium/blob/0585c6beb81cc27e00bd7c9ce47f61bcbc8c9b00/debezium-core/src/main/java/io/debezium/connector/base/ChangeEventQueue.java#L306

**Review by shantanu-sharechat — APPROVED** [review]

**Comment by nancyxu123** [comment]
Hi Jiri, rebased on main.

**Comment by jpechane** [comment]
@nancyxu123 Applied. I've one more hint for improved logging. Please take a look at `io.debezium.util.LoggingContext` and https://logging.apache.org/log4j/2.x/manual/thread-context.html
I believe it would be even better if you use MDC and properly configure logging as it will give you the important metadata without needing them included in all messages.

**Comment by nancyxu123** [comment]
> 

Hi Jiri, thank you so much for the hint! I'll open a public issue to track this work.

**Pull request merged**